### ISSUE TITLE
Validating strings with numbers?

### ISSUE THREAD

**Issue by npearson72** [AngularAgility/AngularAgility#118]
I need to display floats with .00

The problem with the number input type is that it cuts off those decimals, leaving me with numbers like:

4.15
4.1
4

This looks pretty inconsistent in my forms, and is not a good representation of money -- which is what I need.

If I use a standard input field that isn't a number type, it looks fine but does not get validated for numbers.

Is this possible to do? Is there a way to add parseFloat to the number validator?

Thanks

**Comment by johnculviner** [comment]
Hmm, AA just pulls it right off the attribute as you can see here. Might be able to type in the number exactly how you want to see it for this functionality?
https://github.com/AngularAgility/AngularAgility/blob/master/src/formExtensions/directives/ngModel.js#L190

**Comment by npearson72** [comment]
Right, typing the number works, but the problem is when the field gets populated from a server request.

So ideally I can call the server and get a string value of "5.00"  which will appear fine in a text type input. In a number input, I can't really use a string type so I need to pass a float from the server. Then when it gets populated it cuts off the trailing zeros.

What would be great is if AA was indifferent about the type casting. So if I used 5.00 or "5.00" it would 'parseFloat()' and do a min/max comparison.

If the api allows me to create a custom validator for this case, I can work with that too. So if you don't think it makes sense for AA core to do this, would you mind giving me a tip on how to create a custom validator for this?

Thanks so much!

**Comment by johnculviner** [comment]
AA just extends built in validation. If you make your own Angular validators it should work with it provided you register your messages for whatever type you come up with here: https://github.com/AngularAgility/AngularAgility/blob/master/src/formExtensions/provider.js#L225

Making validators as indicated on the website should work:
https://docs.angularjs.org/guide/forms
